packages/office-ui-fabric-react/src/components/Button/ButtonBase.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import {
+  BaseComponent,
+  css,
+  assign,
+  getId,
+  getNativeProps,
+  buttonProperties,
+  anchorProperties
+} from '../../Utilities';
+import { IButtonProps, IButton, ButtonType } from './Button.Props';
+
+export interface IButtonState {
+  labelId?: string;
+  descriptionId?: string;
+  ariaDescriptionId?: string;
+}
+
+export class ButtonBase extends BaseComponent<IButtonProps, IButtonState> implements IButton {
```

Comment:
We can add the `abstract` keyword here to prevent `ButtonBase` from being created on its own, right? Since it's meant to extend from without ever being used on its own. @dzearing?